lib/models/period_wrapper.dart

```diff
@@ -0,0 +1,32 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'period_wrapper.freezed.dart';
+part 'period_wrapper.g.dart';
+
+@freezed
+
+/// Contains all the details associated with a Period
+class Period with _$Period {
+  const factory Period(
+    /// The number of period units: DAY, WEEK, MONTH, YEAR, UNKNOWN
```

Comment:
nitpick but the values are actually lowercased in Dart

Reply:
Ahhh, I had this comment before I made it an enum 🤦‍♂️  Will fix!